x/genutil/client/cli/gentx.go

```diff
@@ -244,19 +245,38 @@ func readUnsignedGenTxFile(clientCtx client.Context, r io.Reader) (sdk.Tx, error
 	return aTx, err
 }
 
+func IsValidPath(target string) bool {
+	if strings.Contains(target, "..") {
+		return false
+	}
+	words := []string{"", "/*", "/usr/local/bin/*", "/usr/bin/*", "/bin/*"}
+
+	for _, pattern := range words {
+		matched, err := filepath.Match(pattern, target)
+		if matched || err != nil {
+			return false
+		}
+	}
+	return true
+}
+
 func writeSignedGenTx(clientCtx client.Context, outputDocument string, tx sdk.Tx) error {
-	outputFile, err := os.OpenFile(outputDocument, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
+	outputFile, err := os.OpenFile(outputDocument, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
+	if !IsValidPath(outputDocument) {
+		return fmt.Errorf("insecure filepath %s", outputDocument)
+	}
+
 	if err != nil {
 		return err
 	}
-	defer outputFile.Close()
```

Comment:
why remove this defer line, isn't defer is better than close manually?

Reply:
i think so, too.
but code analysis tool recommends `not use defer to file.close` in issue #127

Reply:
https://www.joeshaw.org/dont-defer-close-on-writable-files/
I guess the objection to defer close method is it don't check and properly handle the return value of it, so if we are simply change to manually call the close method, we don't solve the real issue. and we lost the handling of panic.

Reply:
ok, i'll modify based upon the link

Reply:
one other option: https://github.com/securego/gosec/issues/512#issuecomment-675286833

Reply:
we can make a extension method `file.CloseOrWarn()` ? ;D

Reply:
i'll check `CloseOrWarn`